### ISSUE TITLE
Security issues

### ISSUE THREAD

**Issue by chesio** [tlovett1/simple-cache#53]
Hi Taylor,

I found two security issues in Simple Cache that I think you should be aware of:

1. Any request to a valid URL with a random query string results in a new cache file being created, because query string is part of `REQUEST_URI` and therefore is being used in file name of the new cache file. This opens door to a possible DOS attacks (sometimes called cache flooding).
1. It is possible to manipulate cache to serve a different page than requested using requests with specially crafted URLs. I can't imagine a scenario when this could be misused, but I believe there are people with far better imagination than me :-)

Both issues can be prevented by one of the following:
1. Not caching any requests with non-empty query string
1. Better sanitization of `REQUEST_URI` in [sc_get_url_path()](https://github.com/tlovett1/simple-cache/blob/master/inc/dropins/file-based-page-cache-functions.php#L100-L105) function.

I personally favour the former, as it generally makes sense to me to avoid caching request that are in essence "dynamic", but I don't know whether your decision to cache requests with query string is deliberate or not; obviously, not caching such requests makes Simple Cache unusable for websites without pretty permalinks.

**Comment by chesio** [comment]
Hi,

It's a bit worrying that these issues are still present in the plugin. Especially the latter issue is quite nasty as an attacker can for example replace content of home page with contents of custom subpage. PoC is very simple: `GET http://example.com/subpage/?/../../`

This works on any WordPress installation with pretty permalinks active. May not work on Windows webservers (not sure how they deal with "?" in directory name).

WordPress will serve page under `http://example.com/subpage/` ignoring the query string part `?/../..`, but Simple Cache will create cache file under `[root-cache-dir]/subpage/?/../../` path which effectively resolves to `[root-cache-dir]/`...

**Comment by msassmann** [comment]
This issue is still present in plugin. Will it be fixed? Would like to use this plugin on many sites because I really like it especially in combination with nginx' `try_files` directive.

**Comment by gagan0123** [comment]
I don't think this plugin is supposed to prevent DOS attacks. It's a "Simple Cache"ing solution, aims to provide the benefits of caching, with just single click install and it does that with ease. Overly complicating things and trying to do everything will simply defeat the purpose of this plugin. As for DOS attacks, there are tools specifically designed to mitigate such attempts. For instance, on Nginx, you can use the rate limits, or can install Fail2Ban, or if you like simplicity, bring your site behind Cloudflare CDN.

**Comment by chesio** [comment]
> I don't think this plugin is supposed to prevent DOS attacks.

It certainly is not supposed to, but also should not make such attacks easier.
